gudid-integration/src/test/java/dev/ikm/tinkar/gudid/integration/GudidGmdnTermsSemanticIT.java

```diff
@@ -0,0 +1,82 @@
+package dev.ikm.tinkar.gudid.integration;
+
+import dev.ikm.tinkar.common.id.PublicIds;
+import dev.ikm.tinkar.coordinate.Coordinates;
+import dev.ikm.tinkar.coordinate.stamp.StampCoordinateRecord;
+import dev.ikm.tinkar.coordinate.stamp.StateSet;
+import dev.ikm.tinkar.coordinate.stamp.calculator.Latest;
+import dev.ikm.tinkar.coordinate.stamp.calculator.StampCalculator;
+import dev.ikm.tinkar.coordinate.stamp.calculator.StampCalculatorWithCache;
+import dev.ikm.tinkar.entity.ConceptRecord;
+import dev.ikm.tinkar.entity.ConceptVersionRecord;
+import dev.ikm.tinkar.entity.Entity;
+import dev.ikm.tinkar.entity.EntityService;
+import dev.ikm.tinkar.entity.EntityVersion;
+import dev.ikm.tinkar.terms.EntityProxy;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class GudidGmdnTermsSemanticIT extends AbstractIntegrationTest {
+    String previousGmdnCode = "";
+
+    /**
+     * Test GudidGmdnTerm Semantics.
+     *
+     * @result Reads content from file and validates Concept of Semantics by calling private method assertLine().
+     */
+    @Test
+    public void testGudidGmdnTermSemantics() throws IOException {
+        String sourceFilePath = "../gudid-origin/target/origin-sources";
+        String errorFile = "target/failsafe-reports/gudid_gmdn_terms_not_found.txt";
+
+        String absolutePath = findFilePath(sourceFilePath, "gmdnTerms.txt");
+        int notFound = processFile(absolutePath, errorFile);
+
+        assertEquals(0, notFound, "Unable to find " + notFound + " Gudid gmdnTerms semantics. Details written to " + errorFile);
+    }
+
+    @Override
+    protected boolean assertLine(String[] columns) {
+        String primaryDi = columns[0];
+        String gmdnName = columns[1];
+        String gmdnDef = columns[2];
+        String gmdnCode = columns[3];
+        String gmdnCodeStatus = columns[4];
+
+        if (!gudidUtility.isDeviceIncluded(primaryDi)) {
+            return true;
+        }
+
+        if (previousGmdnCode.equals(gmdnCode)) { //If current gmdnCode is equals to previous line(s), no need to create Concept then return!
+            return true;
```

Comment:
The file has to be sorted by GMDN code for this logic to work as expected. I think you can remove this condition - the test just has to assert that a GMDN concept exists for every line in the file, excluding filtered device IDs.